File: src/Core/FinTech.Domain/Entities/Transaction.cs
namespace FinTech.Domain.Entities;

public class Transaction
{
    public Guid Id { get; set; }
    public string IdempotencyKey { get; set; } = null!;
    public TransactionType Type { get; set; }
    public decimal Amount { get; set; }
    public TransactionStatus Status { get; set; }
    public Guid? LoanId { get; set; }
    public string Description { get; set; } = null!;
    public DateTime CreatedAt { get; set; }
    public virtual Loan? Loan { get; set; }
}

public enum TransactionStatus
{
    Pending,
    Completed,
    Failed,
}

public enum TransactionType
{
    Disbursement,
    Payment,
    Transfer
}

File: frontend/types/transaction.ts
export type TransactionType = 'Disbursement' | 'Payment' | 'Transfer';

export type TransactionStatus = 'Pending' | 'Completed' | 'Failed';

export interface TransactionResponse {
  id: string;
  idempotencyKey: string;
  type: TransactionType;
  amount: number;
  status: TransactionStatus;
  loanId: string | null;
  description: string;
  createdAt: string;
}

export interface CreateTransactionRequest {
  idempotencyKey: string;
  type: TransactionType;
  amount: number;
  loanId?: string;
  description: string;
}

export interface GetTransactionsParams {
  type?: TransactionType;
  status?: TransactionStatus;
}
